### DISCUSSION TITLE
2.3.210 Upgrade: so-elastalert missing

### DISCUSSION THREAD

**Discussion by sjnhawk** [Security-Onion-Solutions/securityonion#9760] [Unsupported Versions]
Following the upgrade from 2.3.200 to 2.3.210 using soup, so-elastalert is missing. Any thoughts? Screenshots included. I've tried to restart the container and reboot the system with no luck.

so-status:
![image](https://user-images.githubusercontent.com/67609108/217950327-f71735d4-83b9-4207-a746-96d3a18b9237.png)

docker images | grep elastalert:
![image](https://user-images.githubusercontent.com/67609108/217950640-b9e048c4-d713-4bcc-bf9e-1b0dd9833a30.png)

docker logs so-elastalert:
![image](https://user-images.githubusercontent.com/67609108/217950829-dc195518-48b4-4ae5-94de-7e62f099a58c.png)

**Comment by sjnhawk** [comment]
Strange... but it fixed itself after about 2 hours following a reboot. This was the last docker logs so-elastalert:

![image](https://user-images.githubusercontent.com/67609108/217960088-3433d55b-cbc5-4e63-bf2e-02263135469c.png)

I did another reboot to see if it would come back up again, but get the "missing" again and "no such container", after roughly 30 minutes it came back, however I am getting the "502" error on the SOC. I tried to run a so-soc-restart and no joy.

**Comment by cm-ops** [comment]
Are there any clues in the SOC log `/opt/so/log/soc/sensoroni-server.log`? What are your server resources?

**Reply by sjnhawk** [reply]
I checked the log for the day I upgraded and don't see any 502 status codes, all 200. 

Everything is working fine now. Any ideas on why I was getting the "MISSING" for so-elastalert? It eventually remedies itself after time, I've just never seen that happen.

**Reply by sjnhawk** [reply]
I'm starting to think something happened with elastic in general. I tried to get into Kibana and got an error 404 even though everything is showing as "ok" in so-status. Once I ran 'so-kibana-restart' and 'so-elasticsearch-restart' I was able to get into it, however there is no data or dashboards.

This is all I get...

![image](https://user-images.githubusercontent.com/67609108/218915048-9f43e5c1-1c5d-4277-a732-2297fb2e3e42.png)

**Comment by cm-ops** [comment]
For Kibana, try and reload the config by doing the following: `sudo rm /opt/so/state/kibana_*.txt` and `sudo salt-call state.apply kibana.so_savedobjects_defaults -l info queue=True`. Also, what are you resources on this standalone? You could check `dmesg` to see if Elastalert is getting OOM killed. 

If you are still seeing 502 error in SOC, try your query, return a 502 error, check `/opt/so/log/soc/sensoroni-server.log` for the 502 error. Also, verify your cluster is up with `so-elasticsearch-query _cluster/health?pretty` if it is red, look to see what shards are not initialized and started.

**Reply by sjnhawk** [reply]
I ended up just doing a clean install of 2.3.210 and everything is working fine. I have been running 2.3 versions for months and never ran into any issues when I upgraded.

Resources:
Intel Core i3-10100 (4 cores/8 threads)
32GB RAM
4TB WD Red Pro - 7200 RPM

Appreciate the help. Feel free to close this out.
